pinot-common/src/thrift/query.thrift

```diff
@@ -20,6 +20,7 @@ namespace java org.apache.pinot.common.request
 
 struct DataSource {
   1: optional string tableName;
+  2: optional PinotQuery preAggregateGapfillQuery;
```

Comment:
Let's name it `subquery`. We want it to be general in the query syntax

Reply:
Done